apis/kueue/v1alpha2/clusterqueue_types.go

```diff
@@ -243,10 +243,63 @@ type ClusterQueueStatus struct {
 	// clusterQueue and haven't finished yet.
 	// +optional
 	AdmittedWorkloads int32 `json:"admittedWorkloads"`
+
+	// conditions hold the latest available observations of the ClusterQueue
+	// current state.
+	// +optional
+	// +listType=map
+	// +listMapKey=type
+	Conditions []ClusterQueueCondition `json:"conditions,omitempty"`
 }
 
 type UsedResources map[corev1.ResourceName]map[string]Usage
 
+type ClusterQueueCondition struct {
+	// type of condition could be:
+	//
+	// ClusterQueuePending: clusterQueue is accepted but not yet active.
+	// In this state, it can't admit new workloads and its quota can't be borrowed
+	// by other active ClusterQueues in the same cohort.
+	//
+	// ClusterQueueActive: clusterQueue can admit new workloads and its quota
+	// can be borrowed by other ClusterQueues in the same cohort.
+	Type ClusterQueueConditionType `json:"type"`
+
+	// status could be True, False or Unknown.
+	Status corev1.ConditionStatus `json:"status"`
+
+	// lastProbeTime is the last time the condition was checked.
+	// +optional
+	LastProbeTime metav1.Time `json:"lastProbeTime,omitempty"`
+
+	// lastTransitionTime is the last time the condition transit from one status
+	// to another.
+	// +optional
+	LastTransitionTime metav1.Time `json:"lastTransitionTime,omitempty"`
+
+	// reason is a brief reason for the condition's last transition.
+	// +optional
+	Reason string `json:"reason,omitempty"`
+
+	// message is a human-readable message indicating details about last
+	// transition.
+	// +optional
+	Message string `json:"message,omitempty"`
+}
+
+type ClusterQueueConditionType string
+
+const (
+	// ClusterQueuePending indicates clusterQueue is accepted but not yet active.
+	// In this state, it can't admit new workloads and its quota can't be borrowed
+	// by other active ClusterQueues in the same cohort.
+	ClusterQueuePending ClusterQueueConditionType = "Pending"
```

Comment:
No need to have 2 conditions

Just `Active` with status=True/False should be enough.

Reply:
Sounds good.

Reply:
Won't enum type more extensible?

Reply:
If we make `Active` enum type, we will pay more costs to maintain API.
WDYT? @alculquicondor 
https://github.com/kubernetes-sigs/kueue/pull/415#discussion_r999926158

Reply:
I agree with the suggestions by @alculquicondor.

Reply:
Misunderstanding the intension, we should use `metav1.Condition`.